### ISSUE TITLE
Decimal::from_str("0.10").unwrap().log10() does not work

### ISSUE THREAD

**Issue by xiaoyual666** [paupino/rust-decimal#468]
Hi,

This line does not work:
`Decimal::from_str("0.10").unwrap().log10()`

This line works:
`Decimal::from_str("0.1").unwrap().log10()`

Is this a bug?

**Comment by paupino** [comment]
Yes, this looks like a bug. In fact, `0.1`, while not panicking is also giving an incorrect "correct" result. 

From first glance, the reason for this is that it is trying to use an optimized path, however the optimized path is currently for values > 0. It kind of works for `0.1` since it bypasses this path, but `0.10` tries to use it (i.e. treats the mantissa as `10`) which is where the problem starts.

All in all, it shouldn't be too tricky to resolve. It should be something that can be fixed before the next release is cut.
